articles/variables-in-js/index.md

```diff
@@ -0,0 +1,129 @@
+1. Global scope
+2. Local scope
+
+**1. Global scope**
+
+When a variable is *globally scoped*, it means it is available anywhere in your program. 
+Declaring a variable outside a function leads to it being ade to the global scope automatically.
+Global scope uses `var` key word to declare variables.
+
+**2. Local scope**
+
+When variables are declared within a function they become locally scoped, that means they can only be called inside the function.
+Local scope variables are divided into:
+- function scoped variables
+- block scope variables
+
+**a.Function scoped variables**
+
+When you say a variable is function scoped, it means that the variable defined within the function is not accessible from outside the function.
+To define a locally scoped variable we use the `var` keyword.
+
+
+**b.Block scoped variables**
+
+In JavaScript a block is denoted by curly braces, the space between the curly brackets is what is known as a block.
+For example: the `if...else, do...while` and `for` loop statements create blocks.
+
+When a variable is block-scoped, it means that it exists inside the block, it was defined.
+
+`let` and `const` keywords are used in defining the block scope.
+
+### Naming Variables In JavaScript
+Before you start declaring variables, you should first learn how to name them first. Below are some of the rules that one should follow to ensure that your JavaScript code is easy to understand and work with. Variable names;
+- should not have spaces.
+- should begin with an underscore, a dollar sign or a letter.
+- are very case-sensitive. This means `car_TYPE` is treated as an entirely different variable than one named `car_type`
+- must contain only letters, numbers, underscores, or dollar signs.
+- You can't use any JavaScript *reserved words* as a variable name, reserved words include; private, public, enum etc.
+
+### Declaring variables using the keyword `var`
+
+Variables can be declared using the `var` keyword as shown below:
+
+```javascript
+var favouriteThing;
+```
+Upon declaring a variable, you can assign the variable a value such as a string or a number.
+
+Example:
+```javascript
+var favouriteThing;
+favouriteThing = "Icecream!";
+```
+Declaring a variable and at the same time initializing it is also possible using the following syntax;
+```javascript
+var favouriteThing = "Icecream";
+```
+Declaring two or more variables using the same statement is also possible.
+Each declaration is separated by a comma (,).
+
+Example:
+```javascript
+var favouriteThing = "Icecream" ,
+    bestMovie = "Code Eight";
+```
+
+The `var` keyword declares both a function-scoped and globally-scoped variable and can initialize it to a value.
+
+Example:
+```javascript
+var x = 2;
+
+if (x === 2) {
+  var x = 3;
+
+  console.log(x);
+  // expected output: 3
+}
+
+console.log(x);
+//expected output: 3
+```
+
+### Declaring variables using the keyword `let`
+`let` keyword can be used to declare one or more variables. 
```

Comment:
You can remove this line. All the variable declaring keyword can be used to declare one or more variables.